api/v1beta1/gitrepository_types.go

```diff
@@ -26,6 +26,10 @@ import (
 const (
 	// GitRepositoryKind is the string representation of a GitRepository.
 	GitRepositoryKind = "GitRepository"
+	// Represents the go-git git implementation kind
+	GoGitImplementation = "go-git"
+	// Represents the gi2go git implementation kind
+	Git2GoImplementation = "git2go"
```

Comment:
I think it's worth using `libgit2` as the enum value here; still accurate, but looks different to 'go-git'.